vlib/v/parser/tests/anon_fn_return_type.out

```diff
@@ -0,0 +1,6 @@
+vlib/v/parser/tests/anon_fn_return_type.vv:7:13: error: invalid return type `one` for anonymous function
+    5 | ms := &Msg{}
+    6 | 
+    7 | fn (m &Msg) one() { 
```

Comment:
I don't see an anonymous function here?

Reply:
The ms assignment means implicit main. So V parses the fn line as an anonymous function. It's taken from the linked issue.

Reply:
That's a bug in the compiler. We shouldn't be confusing the user.

Reply:
I think we should just require `fn main()` if there's at least one `fn` in the code.

Reply:
I agree

Reply:
> I think we should just require fn main() if there's at least one fn in the code.

I'm not sure that's workable for the parser when a fn comes after a statement.

Reply:
In any case, this pull is correct - it only detects invalid anonymous functions when the parser is already parsing them.

Reply:
I think we should just always require main() :D I never was of not requiring it

Reply:
@medvednikov I think a good solution would be to disallow fn at the start of a top level statement after implicit main is detected. That way the error message won't be confusing. I could do that in another pull. I will update the test here (plus I need to fix allowing mod.Type in the return type).

Reply:
yes that works @ntre

Reply:
Replaced method test case.